src/systems_lock.lua

```diff
@@ -0,0 +1,75 @@
+function plugindef()
+    finaleplugin.RequireSelection = false
+    finaleplugin.Author = "Jacob Winkler"
+    finaleplugin.Copyright = "2022"
+    finaleplugin.Version = "1.0"
+    finaleplugin.Date = "6/28/2022"
+    finaleplugin.AdditionalMenuOptions = [[
+        Systems: Lock Score
+        Systems: Lock Parts
+    ]]
+    finaleplugin.AdditionalUndoText = [[
+        Systems: Lock Score
+        Systems: Lock Parts
+    ]]
+    finaleplugin.AdditionalDescriptions = [[
+        Lock All Systems (Score)
+        Lock All Systems (Parts)
+    ]]
+    finaleplugin.AdditionalPrefixes = [[
+        lock = "Score"
+        lock = "Parts"
+    ]]
+    return "Systems: Lock All", "Systems: Lock All", "Lock All Systems (Score & Parts)"
+end
+
+    lock = lock or "All"
+
+function formatting_systems_lock(lock)
+    lock = lock or "All"
+
+    local lock_score = true
+    local lock_parts = true
+
+    if lock == "Score" then
+        lock_parts = false
+    elseif lock == "Parts" then
+        lock_score = false
+    end
+
+    function systems_lock()
```

Comment:
This looks like a good function to move to the top-level scope (see above comment on why).